### ISSUE TITLE
Keyboard nav on Safari broken for scrolling list of checkboxes

### ISSUE THREAD

**Issue by jhpedemonte** [carbon-design-system/carbon#6444]
## Environment

> Operating system

MacOS 10.15.5

> Browser

Safari 13.1.1

> Assistive technology used to verify

Keyboard navigation

## Detailed description

> What version of the Carbon Design System are you using?

10.15

> What did you expect to happen?

Able to `TAB` through a list of checkbox items and see currently focused item.

> What happened instead?

In Safari, the list scrolling goes haywire and currently focused item is not shown.

## Steps to reproduce the issue

1. Load https://codesandbox.io/s/red-cookies-vmkou?file=/src/index.js in Safari
2. Click on each heading and then try to `TAB` through each list
3. Notice that tabbing in the first two lists works fine for some of the first items, but then list scrolls excessively and the focused item is no longer shown.
4. Notice that it works as expected for the 3rd list, where `position: relative` has been added to `.bx--checkbox-wrapper`

**Comment by jhpedemonte** [comment]
I have no idea why this happens only in Safari. Only solution we found was to add `position: relative` to the parent of the `<input>` (which is `position: absolute`).

**Comment by dakahn** [comment]
Is this a bug with checkboxes in groups on Safari _in general_ or specific to our implementation?

**Comment by dakahn** [comment]
It appears as though keyboard functionality for _all_ checkboxes (not just those in scrolling lists) is broken in Safari. Thanks for calling this out. 👍

**Comment by sjarva** [comment]
@dakahn apparently this is a "feature" on Safari, as they have a setting in Safari > Preferences > Advanced: Accessibility: [checkbox] Press tab to highlight each item on web page

<img width="773" alt="Safari Preferences, Press tab to highlight each item on web page checkbox" src="https://user-images.githubusercontent.com/1133238/95658751-61f3df80-0b25-11eb-871f-c70e1bc4ccb1.png">

If you haven't checked it, by using Tab Safari will only highlight text and lists that can get focus. By using Option + Tab you will see the focus in all controls (buttons, checkboxes etc).

@avigil06 wrote [a blog post about trying to code around this in Safari](https://medium.com/@avigil06/keyboard-accessible-radios-and-checkboxes-in-safari-5be5ad7fd399).

There is also an OS level control to change this (System Preferences > Keyboard > Shortcuts:

[radio button on bottom] Full Keyboard Access: in windows and dialogs, press Tab to move keyboard focus between:

[default option] Text boxes and lists only
[option] All controls.

<img width="620" alt="Mac OS System preferences Full keyboard access setting" src="https://user-images.githubusercontent.com/1133238/95658767-88197f80-0b25-11eb-8066-1d42a25b9aa2.png">

No idea why Mac OS and Safari are making these the default setting, but I guess that users who use Mac, Safari and want to navigate by keyboard, know this 🤷 But, I do agree with @avigil06 that these behaviours should be consistent across all browsers.

**Comment by tw15egan** [comment]
@dakahn based on @sjarva's comments, is there any action we want to take here?

**Comment by joshblack** [comment]
bump @dakahn when you get a chance

**Comment by dakahn** [comment]
I think we're all clear here. Something to note moving forward. Great find @sjarva! Thank you so much!
